src/npctalk.cpp

```diff
@@ -1940,6 +1940,32 @@ void talk_effect_fun_t::set_u_buy_item( const std::string &item_name, int cost,
     };
 }
 
+void talk_effect_fun_t::set_u_sell_item( const std::string &item_name, int cost, int count )
+{
+    function = [item_name, cost, count]( const dialogue &d ) {
+        npc &p = *d.beta;
+        player &u = *d.alpha;
+        item old_item( item_name, calendar::turn );
+        for( int i = 0; i < count; i++ ) {
+            int item_at = u.inv.position_by_type( item_name );
+            if( item_at == INT_MIN ) {
+                popup( _( "You don't have a %1$s!" ), old_item.tname() );
+                return;
+            }
+            old_item = u.i_rem( item_at );
+            p.i_add( old_item );
+        }
+        if( count == 1 ) {
+            //~ %1%s is the NPC name, %2$s is an item
+            popup( _( "You give %1$s a %2$s" ), p.name, old_item.tname() );
+        } else {
+            //~ %1%s is the NPC name, %2$d is a number of items, %3$s are items
+            popup( _( "You give %1$s %2$d %3$s" ), p.name, count, old_item.tname() );
+        }
+        u.cash -= cost;
```

Comment:
Shouldn't this be `u.cash += cost;` instead? Unless costs for items are to be set as negative values, of course.

Reply:
Fixed.